### PR TITLE
[RLlib] Offline Type Annotations

### PR DESCRIPTION
**Pull request by michaelzhiluo**
## Why are these changes needed?

Needs more type annotations

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: rllib/offline/input_reader.py [hunk 1]
```diff
@@ -25,7 +27,7 @@ def next(self):
         raise NotImplementedError
 
     @PublicAPI
-    def tf_input_ops(self, queue_size=1):
+    def tf_input_ops(self, queue_size: int = 1) -> TensorStructType:
```
Comment (ericl) [inline rllib/offline/input_reader.py:30]: Dict[str, TensorType]

(this isn't a gym space so shouldn't be tensor struct)

File: rllib/utils/types.py [hunk 2]
```diff
@@ -80,6 +80,9 @@
 # Some kind of sample batch.
 SampleBatchType = Union["SampleBatch", "MultiAgentBatch"]
 
+# Represents a generic Serial type
+SerializedType = Any
```
Comment (ericl) [inline rllib/utils/types.py:84]: Instead of adding this let's just use `str`

File: rllib/offline/json_writer.py [hunk 3]
```diff
@@ -72,7 +74,7 @@ def write(self, sample_batch):
             len(data), f,
             time.time() - start))
 
-    def _get_file(self):
+    def _get_file(self) -> SerializedType:
```
Comment (ericl) [inline rllib/offline/json_writer.py:77]: Should be returning a file type.

File: rllib/offline/json_writer.py [hunk 4]
```diff
@@ -94,15 +96,16 @@ def _get_file(self):
...
     if compress and compression_supported():
         return str(pack(v))
     elif isinstance(v, np.ndarray):
         return v.tolist()
     return v
 
 
-def _to_json(batch, compress_columns):
+def _to_json(batch: SampleBatchType,
+             compress_columns: List[str]) -> SerializedType:
```
Comment (ericl) [inline rllib/offline/json_writer.py:108]:
```suggestion
             compress_columns: List[str]) -> str:
```

Please use `str` here and elsewhere.

**Review by ericl — CHANGES_REQUESTED** [review]
Made a few comments

File: rllib/offline/json_writer.py (hunk 3 above)
Comment (michaelzhiluo) [reply rllib/offline/json_writer.py:77]: I put FileType=Any in typing. Bc the actual class of files is called: _io.TextIOWrapper

**Comment by michaelzhiluo** [comment]
Aight @ericl pushed revisions

**Review by ericl — APPROVED** [review]

**Comment by sven1977** [comment]
@michaelzhiluo Could you check the tests once more? They are all failing. Thx! :)

**Comment by michaelzhiluo** [comment]
@sven1977 It's the same issue that was mentioned yesterday! `cannot import name 'RolloutWorker' from 'ray.rllib.evaluation.rollout_worker'`

**Comment by michaelzhiluo** [comment]
@sven1977 Fixed

**Pull request merged**